src/preprocessing/passes/symmetry_breaker.cpp

```diff
@@ -0,0 +1,119 @@
+/*********************                                                        */
+/*! \file symmetry_breaker.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **  Paul Meng, Andrew Reynolds
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2018 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Symmetry breaker module
+ **/
+
+#include "preprocessing/passes/symmetry_breaker.h"
+
+using namespace std;
+
+namespace CVC4 {
+namespace preprocessing {
+namespace passes {
+
+Node SymmetryBreaker::generateSymBkConstraints(const vector<vector<Node>>& parts)
+{
+  Node sbConstraint = d_trueNode;
+
+  for (const vector<Node>& part : parts)
+  {
+    if (part.size() >= 2)
+    {
+      Kind kd = getOrderKind(part[0]);
+
+      if (kd != kind::EQUAL)
+      {
+        for (unsigned int i = 0; i < part.size() - 1; ++i)
+        {
+          // Generate less than or equal to constraints: part[i] <= part[i+1]
+          Node constraint =
+              NodeManager::currentNM()->mkNode(kd, part[i], part[i + 1]);
+          sbConstraint = NodeManager::currentNM()->mkNode(
```

Comment:
Can you add constraint to a vector and make a single constraint at the end? It is inefficient to make all these intermediate nodes, same with below.

Reply:
Sure. 
But do I need to check if the vector constraints is empty or not before calling mkNode(kind::AND, constraints)? Or mkNode will take care of the empty case and also the single element?

Reply:
Unfortunately, no, this is done in various places in the code, see: https://github.com/ajreynol/CVC4/blob/impRealIntParsePrint/src/theory/datatypes/theory_datatypes.cpp#L2185
for an example.  We're planning to make a utility for this at some point, but you should just write these 6 lines in your code for now.